server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -437,10 +435,9 @@ static Settings aggregateIndexSettings(ClusterState currentState, CreateIndexClu
          * that will be used to create this index.
          */
         MetaDataCreateIndexService.checkShardLimit(indexSettings, currentState);
-        if (indexSettings.getAsBoolean(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true) == false) {
-            DEPRECATION_LOGGER.deprecatedAndMaybeLog("soft_deletes_disabled",
-                "Creating indices with soft-deletes disabled is deprecated and will be removed in future Elasticsearch versions. " +
-                "Please do not specify value for setting [index.soft_deletes.enabled] of index [" + request.index() + "].");
+        if (IndexSettings.INDEX_SOFT_DELETES_SETTING.get(indexSettings) == false) {
```

Comment:
should we still allow `IndexSettings.INDEX_SOFT_DELETES_SETTING` to be set to `false` if `SETTING_INDEX_VERSION_CREATED` is before V_8_0_0. This will allow shrink/split of non-soft-deleted indices continue to work. Also, a test for this would be good

Reply:
fixed and added a test in https://github.com/elastic/elasticsearch/pull/51122/commits/0ecee877a4c92beef7bfac4735e68056555b48e3.